table/block.cc

```diff
@@ -282,14 +293,56 @@ bool BlockIter::ParseNextKey() {
     }
 
     value_ = Slice(p + non_shared, value_length);
-    while (restart_index_ + 1 < num_restarts_ &&
-           GetRestartPoint(restart_index_ + 1) < current_) {
+    uint32_t next_restart_point = 0;
+    bool next_inited = false;
+    while (restart_index_ + 1 < num_restarts_ && (next_inited = true) &&
+           (next_restart_point = GetRestartPoint(restart_index_ + 1)) <
+               current_) {
       ++restart_index_;
     }
+    if (value_delta_encoded_) {
+      assert(value_length == 0);
+      DecodeCurrentValue(next_inited, next_restart_point, p, non_shared);
+    }
     return true;
   }
 }
 
+// The fomrat:
+// restart_point   0: k, v (offset, size), k, v (size), ..., k, v (size)
+// restart_point   1: k, v (offset, size), k, v (size), ..., k, v (size)
+// ...
+// restart_point n-1: k, v (offset, size), k, v (size), ..., k, v (size)
+// where, k is key, v is value, and its encoding is in pranthesis.
```

Comment:
Also here "pranthesis"

Reply:
@ot yeah you found the description, and as it says:
> The format of each key is (shared_size, non_shared_size, shared, non_shared)

so the block handle size (i.e., the value size) is not encoded in the new format.